File: python/app/elasticsearch_client.py
# Documentation: https://elasticsearch-py.readthedocs.io/
from elasticsearch import AsyncElasticsearch
from . import config

"""Elasticsearch client and utilities."""

HOST: str = config.read("ELASTICSEARCH_HOST")
API_KEY: str = config.read("ELASTICSEARCH_API_KEY")


def get_client():
    client = AsyncElasticsearch(hosts=HOST, api_key=API_KEY, verify_certs=False)
    return client

File: node/src/core/elasticsearch_client.ts
// https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/introduction.html
import { Client } from "@elastic/elasticsearch";
import { readEnvironmentVariable } from "./config";

const getClient = () =>
  new Client({
    node: readEnvironmentVariable("ELASTICSEARCH_HOST"),
    requestTimeout: 1 * 60000, // 1min timeout
    auth: {
      apiKey: readEnvironmentVariable("ELASTICSEARCH_API_KEY"),
    },
    tls: {
      // https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/client-connecting.html
      rejectUnauthorized: false,
    },
  });

export { getClient };
